torchsynth/globals.py

```diff
@@ -30,12 +34,30 @@ def __init__(
         self.sample_rate = sample_rate
         self.buffer_size = buffer_size
 
+        # Default control rate to 100 times less than audio rate
+        if control_rate is None:
+            self.control_rate = (self.sample_rate // 100).clone().detach()
+        else:
+            assert control_rate.ndim == 0
+            self.control_rate = control_rate
+
+        # Buffer size for control signals
+        self.control_buffer_size = (
+            torch.ceil((buffer_size / sample_rate * self.control_rate))
```

Comment:
If the above conditions hold then we assert control_rate * buffer_size // sample_rate = this ceil value?

Reply:
Yes sounds good

Reply:
hmm, so this gets weird with some buffer sizes -- I don't think we can make this assertion. Ex: say we have a buffer size of 4096, sr of 48k, and a control rate of 480.  `ceil(480 * 4096 /48000) = 41`, `control_rate * buffer_size // sample_rate = 40`

I think this will be fine to leave as is. The upsample operation works by upsampling to buffer size number of samples, so the number of samples that goes in doesn't really matter. If anything actually we should round and not ceil so we get the closest number of samples.